src/analyses/base.ml

```diff
@@ -764,9 +764,11 @@ struct
       let warn_type = (M.EventType.Behavior (M.BehaviorEvent.Undefined (M.UndefinedBehavior.NullPointerDereference))) in
         match (eval_rv a gs st n) with
         | `Address adr -> 
-            if AD.is_null adr then M.warn_each (M.LogEvent.must warn_type)
-            else if AD.may_be_null adr then M.warn_each (M.LogEvent.may warn_type);
-          do_offs (AD.map (add_offset_varinfo (convert_offset a gs st ofs)) adr) ofs
+          if AD.is_null adr 
+            then M.warn_each (M.LogEvent.must warn_type)
+          else if AD.may_be_null adr 
+            then M.warn_each (M.LogEvent.may warn_type);
+            do_offs (AD.map (add_offset_varinfo (convert_offset a gs st ofs)) adr) ofs
```

Comment:
The `do_offs` is not only executed when the `else if` is executed, right? So the indentation is a bit misleading.

Reply:
No it always is executed, Ill add brackets to make it more clear!